turbopack/crates/turbopack-core/src/chunk/mod.rs

```diff
@@ -174,14 +173,14 @@ pub trait OutputChunk: Asset {
     NonLocalValue,
 )]
 pub enum ChunkingType {
-    /// Module is placed in the same chunk group and is loaded in parallel. It
-    /// doesn't become an async module when the referenced module is async.
-    #[default]
-    Parallel,
-    /// Module is placed in the same chunk group and is loaded in parallel. It
-    /// becomes an async module when the referenced module is async.
-    // TODO make inherit_async a separate field
-    ParallelInheritAsync,
+    /// The referenced module is placed in the same chunk group and is loaded in parallel.
+    Parallel {
+        /// Whether the parent module becomes an async module when the referenced module is async.
+        inherit_async: bool,
+        /// Whether the referenced module is executed always immediately before the parent module
```

Comment:
when would this not be the case?

```
// module a.js
await something()

// module b.js
import * as a from 'a'
```

shouldn't module b always be marked async?  why do we need this flag?

im guessing this means i don't exactly understand how this enum is used :)

Reply:
That is a very good question question. For ESM imports, this is definitely always true.

But off the top my head, I also cannot give you an example where you'd want `inherit_async: false`...

Reply:
Ah, I guess one example would be a `require`, which could return just a `Promise` if the required module has a top-level-await.
(Though this is pretty underspecified and probably not something that should be relied on across the ecosystem).